### PR TITLE
Upgrading to Polkadot v0.9.38 and relevant dependencies

### PR DESCRIPTION
**Pull request by chrisli30**

### CODE REVIEW

**Review by chrisli30 — COMMENTED** [review]
I had to create a PR for the comments.

**Comment by chrisli30** [comment]
Regarding the xcmWeight of timeAutomation, let‘s use destWeightLimit: Unlimited for this version, and a parameter interface in the next one.

https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fwss.api.moonriver.moonbeam.network#/extrinsics

CleanShot 2023-05-07 at 18 29 33

CleanShot 2023-05-07 at 18 29 55

**Comment by chrisli30** [comment]
@imstar15 Don’t forget to separate migration removal code from this PR.

File: node/src/chain_spec/turing.rs [hunk 1]
```diff
@@ -16,11 +17,16 @@ use turing_runtime::{
...
 	VersionedMultiLocation,
-	VersionedMultiLocation::V1,
+	VersionedMultiLocation::V3,
 };
+// use xcm::{
+// 	latest::prelude::*,
+// 	v2::Junction::{PalletInstance, Parachain},
+// 	VersionedMultiLocation,
+// 	VersionedMultiLocation::V2,
+// };
```
Comment (chrisli30) [inline node/src/chain_spec/turing.rs:29]: Remove unused code.

File: node/src/command.rs [hunk 2]
```diff
@@ -382,8 +383,11 @@ pub fn run() -> Result<()> {
 					)
 					.map_err(|e| format!("Error: {:?}", e))?;
 
+					// runner
+					// 	.async_run(|config| Ok((cmd.run::<Block, Executor>(config), task_manager)))
+
 					runner
-						.async_run(|config| Ok((cmd.run::<Block, Executor>(config), task_manager)))
+						.async_run(|config| Ok((cmd.run::<Block, Executor>(), task_manager)))
```
Comment (chrisli30) [inline node/src/command.rs:390]: This part needs cleanup.

File: pallets/automation-time/src/lib.rs [hunk 3]
```diff
@@ -1122,14 +1121,18 @@ pub mod pallet {
...
-					});
+					// Self::deposit_event(Event::AutoCompoundDelegatorStakeFailed {
+					// 	task_id,
+					// 	error_message: Into::<&str>::into(e).as_bytes().to_vec(),
+					// 	error: e,
+					// });
+					// (
+					// 	<T as Config>::WeightInfo::run_auto_compound_delegated_stake_task(),
+					// 	Some(e.error),
+					// )
```
Comment (chrisli30) [inline pallets/automation-time/src/lib.rs:1132]: This part of code needs attention.

File: pallets/xcmp-handler/src/benchmarking.rs [hunk 4]
```diff
@@ -38,7 +38,7 @@ benchmarks! {
 		let location = MultiLocation::new(1, X1(Parachain(1000)));
 		let versioned_location: VersionedMultiLocation = location.clone().into();
 		let xcm_data =
-			XcmAssetConfig { fee_per_second: 100, instruction_weight: 1_000, flow: XcmFlow::Normal };
+			XcmAssetConfig { fee_per_second: 100, instruction_weight: Weight::from_parts(1_000, 0), flow: XcmFlow::Normal };
```
Comment (chrisli30) [inline pallets/xcmp-handler/src/benchmarking.rs:41]: Remember to write migration for xcmConfig, for u64 -> Weight.

File: pallets/xcmp-handler/src/lib.rs [hunk 5]
```diff
@@ -333,35 +332,33 @@ pub mod pallet {
...
+				WithdrawAsset::<<T as pallet::Config>::RuntimeCall>(local_asset.clone().into()),
+				DepositAsset::<<T as pallet::Config>::RuntimeCall> {
 					assets: Wild(All),
-					max_assets: 1,
 					beneficiary: MultiLocation { parents: 1, interior: X1(Parachain(para_id)) },
 				},
 			]);
 
+			// TODO: Calculate proof size
+			let proof_size = 5000u64;
```
Comment (chrisli30) [inline pallets/xcmp-handler/src/lib.rs:343]: Delete line 342 343

File: runtime/neumann/src/xcm_config.rs [hunk 6]
```diff
@@ -268,85 +277,127 @@ pub type Trader =
...
+// 	type XcmReserveTransferFilter = Nothing;
+// 	type Weigher = FixedWeightBounds<UnitWeightCost, RuntimeCall, MaxInstructions>;
+// 	type UniversalLocation = UniversalLocation;
+// 	type RuntimeOrigin = RuntimeOrigin;
+// 	type RuntimeCall = RuntimeCall;
+
+// 	const VERSION_DISCOVERY_QUEUE_SIZE: u32 = 100;
+// 	// ^ Override for AdvertisedXcmVersion default
+// 	type AdvertisedXcmVersion = pallet_xcm::CurrentXcmVersion;
+// }
```
Comment (chrisli30) [inline runtime/neumann/src/xcm_config.rs:342]: Remember to delete commented code.

File: runtime/neumann/src/xcm_config.rs [hunk 7]
```diff
@@ -268,85 +277,127 @@ pub type Trader =
...
+	type RuntimeCall = RuntimeCall;
 	const VERSION_DISCOVERY_QUEUE_SIZE: u32 = 100;
-	// ^ Override for AdvertisedXcmVersion default
 	type AdvertisedXcmVersion = pallet_xcm::CurrentXcmVersion;
+	type Currency = Balances;
+	type CurrencyMatcher = ();
+	type TrustedLockers = ();
+	type SovereignAccountOf = LocationToAccountId;
+	type MaxLockers = ConstU32<8>;
+	// TODO pallet-xcm weights
```
Comment (chrisli30) [inline runtime/neumann/src/xcm_config.rs:369]: Remember to delete commented code.

**Review by imstar15 — APPROVED** [review]

**Pull request merged**